geode-dunit/src/main/java/org/apache/geode/test/junit/rules/ServerStarterRule.java

```diff
@@ -167,6 +171,23 @@ public ServerStarterRule withRegion(RegionShortcut type, String name) {
     return this;
   }
 
+  public Region createRegion(RegionShortcut type, String name,
+      Consumer<RegionFactory> regionFactoryConsumer) {
+    RegionFactory factory = getCache().createRegionFactory(type);
+    regionFactoryConsumer.accept(factory);
+    return factory.create(name);
+  }
+
+  public Region createPRRegion(String name, Consumer<RegionFactory> regionFactoryConsumer,
+      Consumer<PartitionAttributesFactory> prAttributesFactory) {
```

Comment:
should this parameter named by changed to "partitionFactoryConsumer"? You could also add Attributes to its name but I don't think it is needed. The current name made me think it was a factory instead of a consumer